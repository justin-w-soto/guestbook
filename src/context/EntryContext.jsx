import React from 'react'
import { useContext, createContext, useState } from 'react'

const EntryContext = createContext()

const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([])

    return ( <EntryContext.Provider value={{ entries, setEntries }}>{ children }</EntryContext.Provider>)
}

const useEntries = () => {
  
  const context = useContext(EntryContext) 

  if (context === undefined) {
    throw new Error('useEntries must be inside of EntryContext Provider')
  }

  return context
}

export { EntryProvider, useEntries }
