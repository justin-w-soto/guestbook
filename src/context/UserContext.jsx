import React from 'react'
import { useContext, createContext, useState }from 'react'

    const UserContext = createContext()

    const UserProvider = ({ children }) => {
        const [user, setUser] = useState('')
        return (
        <UserContext.Provider 
        value={{user, setUser}}
        >
            { children }
        </UserContext.Provider>)
    }

    // MAKE A CUSTOM HOOK 
    const useMyHook = () => {
        const context = useContext(UserContext)
        
        // if (context === undefined) {
        //     throw new Error('useMyHook needs to be called inside of a UserContext Provider')
        // }
        return context
    }

   
export { UserProvider, useMyHook }
