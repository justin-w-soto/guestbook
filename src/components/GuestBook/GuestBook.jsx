import React from 'react'
import { useState } from 'react'
import { useHistory } from 'react-router-dom'
import { useEntries } from '../../context/EntryContext'
import { useUser } from '../../context/UserContext'

export const GuestBook = () => {
const [name, setName] = useState('')
const [guestEntry, setGuestEntry] = useState('')
const { entries, setEntries } = useEntries()
const { user } = useUser('user')
const history = useHistory()

const handleSubmit = (e) => {
    e.preventDefault()
    updateName()
  }

const updateName = () => {
  if (!guestEntry) return 
   setEntries([...entries, {  name, message: guestEntry }]), setGuestEntry('')
}  

const redirectLogin = () => {
  let path = '/login';
  history.replace(path);

}

const msg = user ? `Thanks for the note ${user}.` : 'Sign the Book!'

    return (

      <>
        <h1>{msg}</h1>
        <form onSubmit={handleSubmit}>
          {user ? null : guestEntry}
            <label htmlFor='guestEntry'>Guest Entry</label>

            <textarea 
            id="guestEntry" 
            value={guestEntry} 
            placeholder="Your Note!" 
            onChange={(e) => setGuestEntry(e.target.value)} 
            />

            <button className="button" type="submit">Sign</button>
            {user && (
            <button 
              type="button" 
              onClick={redirectLogin}
              >Not {user} ?
            </button>
              )}
          </form>
        </>
    )
}