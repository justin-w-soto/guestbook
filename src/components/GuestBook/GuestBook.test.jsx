import React from "react";
import { render } from '@testing-library/react';
import { GuestBook } from './GuestBook'
import { UserProvider } from '../../context/UserContext'
import { EntryProvider } from '../../context/EntryContext'

it('should render the guestbook view', ()=>{
    const { container } = render(
    <UserProvider>
        <EntryProvider>
            < GuestBook />
        </EntryProvider>
    </UserProvider>
    );

    expect(container).toMatchSnapshot();
});
