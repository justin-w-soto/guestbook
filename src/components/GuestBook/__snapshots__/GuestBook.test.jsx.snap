// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the guestbook view 1`] = `
<div>
  <h1>
    Sign the Book!
  </h1>
  <form>
    
    <label
      for="guestEntry"
    >
      Guest Entry
    </label>
    <textarea
      id="guestEntry"
      placeholder="Your Note!"
    />
    <button
      class="button"
      type="submit"
    >
      Sign
    </button>
    
  </form>
</div>
`;
